### PR TITLE
DA-4-Add-redux-toolkit-for-user-reducer-slice

### PR DESCRIPTION
**Pull request by afekTheMiniLearner**

### CODE REVIEW

File: client/src/store/slices/users.slice.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  reducers: {
+    initUsersList(state, action) {
+      state.usersList = action.payload.usersList;
+    },
+    initUser(state, action) {
+      state.user = action.payload.user;
+    },
+    changeUser(state, action) {
+      const { id } = action.payload;
+      const { user } = state.usersList.find((u) => u.id === id);
```
Comment (hdriel) [inline client/src/store/slices/users.slice.js:18]: why destruction user from find response ? 
it's not should be: `const user  = state.usersList.find((u) => u.id === id);`

**Review by hdriel — CHANGES_REQUESTED** [review]

File: client/src/store/slices/users.slice.js (hunk 1 above)
Comment (afekTheMiniLearner) [reply client/src/store/slices/users.slice.js:18]: ok will change that, haven't noticed sir

File: client/src/store/slices/users.slice.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    user: null,
+  },
+  reducers: {
+    initUsersList(state, action) {
+      state.usersList = action.payload.usersList;
+    },
+    initUser(state, action) {
+      state.user = action.payload.user;
+    },
+    changeUser(state, action) {
```
Comment (hdriel) [inline client/src/store/slices/users.slice.js:16]: why and for what this action ? 
probably you could get the user from server request , and not from the userList
also, probably you will fetch userList by pagination or specific users, and not all users on you db 
the userList is for display user list mode, not to describe you all users in your db 
so, the action to select user by user id is not realistic

Comment (hdriel) [reply client/src/store/slices/users.slice.js:16]: so, remove this action please

Comment (afekTheMiniLearner) [reply client/src/store/slices/users.slice.js:16]: ok removed

**Pull request merged**